plone/app/discussion/browser/moderation.py

```diff
@@ -13,69 +15,119 @@
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from Products.statusmessages.interfaces import IStatusMessage
 from zope.event import notify
-from zope.interface import alsoProvides
+
+
+# Translations for generated values in buttons
+# States
+_('comment_pending', default='pending')
+# _('comment_approved', default='published')
+_('comment_published', default='published')
+_('comment_rejected', default='rejected')
+_('comment_spam', default='marked as spam')
+# Transitions
+_('Recall')
+_('Approve')
+_('Reject')
+_('Spam')
+
+
+class TranslationHelper(BrowserView):
+
+    def translate(self, text=""):
+        return _(text)
+
+    def translate_comment_review_state(self, rs):
+        return _("comment_" + rs, default=rs)
 
 
 class View(BrowserView):
-    """Comment moderation view.
-    """
+    """Show comment moderation view."""
+
     template = ViewPageTemplateFile('moderation.pt')
     try:
         template.id = '@@moderate-comments'
     except AttributeError:
         # id is not writeable in Zope 2.12
         pass
 
+    def __init__(self, context, request):
+        super(View, self).__init__(context, request)
```

Comment:
I would avoid customizing the `__init__` of the BrowserView, especially with mutable objects (the `transitions` list).
It makes hard to understand what happens under the hood and might lead to unexpected results.